taxonomy-manager-common/src/main/java/com/digirati/taxman/common/rdf/identity/IdResolver.java

```diff
@@ -0,0 +1,53 @@
+package com.digirati.taxman.common.rdf.identity;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+public class IdResolver {
+
+    // This does not need to be ever instanciated
+    private IdResolver() {
+    }
+
+    public static Optional<UUID> resolve(String uri, Pattern pattern) {
+        if (uri == null)
+            return Optional.empty();
+
+        try {
+            var parsedUri = new URI(uri);
+            return resolve(parsedUri, pattern);
+        } catch (URISyntaxException ignored) {
+            return Optional.empty();
+        }
+    }
+    public static Optional<UUID> resolve(URI uri, Pattern pattern) {
+        if (uri == null || pattern == null) {
```

Comment:
Why are we calling this if `uri` or `pattern` are null?

Reply:
So all callers, current and future, can just call and get the answer.

- Can you try and resolve an id from this uri using this pattern? I just grabbed them and I have no idea what they are.
- Sure... nope, no id. You gave me lots of nulls.

Alternatively every caller of this method needs to do their checks, in the end leading to possible code duplication.

Reply:
> Alternatively every caller of this method needs to do their checks, in the end leading to possible code duplication.

This is the correct approach. Allowing null everywhere, while having to test if you've been passed a `null`, is just nonsense.

Reply:
If anything, it should throw an exception on `null` values. Not return a semantically correct result. Passing a null is a programmer error.